scipy/spatial/qhull.pyx

```diff
@@ -1558,6 +1611,64 @@ class _QhullUser(object):
         self._qhull.add_points(points)
         self._update(self._qhull)
 
+    def _update_halfspaces(self, qhull):
+        self.halfspaces = qhull.get_points()
+        self.ndim = self.halfspaces.shape[1] - 1
+        self.nineq = self.halfspaces.shape[0]
+        self.min_bound = self.halfspaces.min(axis=0)
+        self.max_bound = self.halfspaces.max(axis=0)
+
+    def _add_halfspaces(self, halfspaces, restart=False):
```

Comment:
Code duplication.
- Use the same internal `_add_points(self, points, old_points, restart=False)` method for updating points and halfspaces.
- Make the appropriate call to `_add_points` from the `add_points` or `add_halfspaces` of the subclass.
- Override `_update` in `HalfspaceIntersection` instead of adding `_update_halfspaces` and `self._is_halfspaces`
- You can copy the `add_points` docstring in